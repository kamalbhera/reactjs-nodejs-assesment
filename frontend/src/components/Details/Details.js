import React from 'react';
import PropTypes from 'prop-types';

import './Details.scss';

const getSalaryString = (salary) => {
	const formatter = new Intl.NumberFormat('en-GB', {
		style: 'currency',
		currency: 'GBP',
	});
	return formatter.format(salary).replace('.00','');
};

const Details = ({ salary, type, location, company }) => {
	const salaryString = getSalaryString(salary);
	return (
		<div className="details">
			<div>{ location }</div>
			<div>{ salaryString }</div>
			<div>{ type }</div>
			<div>{ company }</div>
		</div>
	);
};

Details.propTypes = {
	salary: PropTypes.number,
	type: PropTypes.string,
	location: PropTypes.string,
	company: PropTypes.string
};

export default Details;
