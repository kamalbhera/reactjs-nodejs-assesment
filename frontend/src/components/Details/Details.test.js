import React from 'react';
import { shallow } from 'enzyme';

import Details from './Details';

it('renders Details component', () => {
	const props = {
		location: 'London',
		salary: 50000,
		type: 'Permanent',
		company: 'Acme'
	};
	const result = shallow(<Details { ...props }/>);
	const html = <div className="details"><div>London</div><div>Â£50,000</div><div>Permanent</div><div>Acme</div></div>;
	expect(result).toMatchElement(html);
	expect(result).toMatchSnapshot();
});
