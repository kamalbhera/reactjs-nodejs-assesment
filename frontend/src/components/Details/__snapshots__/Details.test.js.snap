// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Details component 1`] = `
<div
  className="details"
>
  <div>
    London
  </div>
  <div>
    Â£50,000
  </div>
  <div>
    Permanent
  </div>
  <div>
    Acme
  </div>
</div>
`;
